### ISSUE TITLE
[feature request] Add support for .eslintrc directory

### ISSUE THREAD

**Issue by ctjlewis** [eslint/eslint#13578]
**The version of ESLint you are using.** 
`latest`

**The problem you want to solve.**
 I want to be able to store an eslint config as its own standalone git repository, so it can be added to other repositories with `git submodule add` and managed independently. Git repos must be folders.

**Your take on the correct solution to problem.** 
`.eslintrc` check for a directory `./` should also detect a `.eslint/` or `.eslintrc/` directory, and then use the default `.eslintrc*` logic to select a config `.eslintrc*` in that folder and apply it with respect to the parent folder `./`. 

**Are you willing to submit a pull request to implement this change?** 
Yes.

**Comment by anikethsaha** [comment]
Thanks for the issue. 

Though this is not 100% what you are looking for, maybe `--config` can be helpful like give `--config .eslintrc/.eslintrc` but it won't work completely as you want. 

so about the proposal, this an interesting one, I have seen many tools supporting this. It would be helpful for a better project structure.

You need to submit an RFC for detailed design. But we can discuss here about the interest and queries from others.

**Comment by aladdin-add** [comment]
to clarify:
* eslint <7.0 does ignores `.*` files by default
* as of eslint v7.0, it does not ignore `.eslintrc*` by default

and, it will be changed in eslint v8, as you expected.  :)

previous discussion:
* https://github.com/eslint/eslint/issues/10341
https://github.com/eslint/rfcs/pull/55
* https://github.com/eslint/rfcs/pull/9

**Comment by aladdin-add** [comment]
atm, you can add a `!.eslintrc/` in your `.eslintignore`

**Comment by kaicataldo** [comment]
I don't think core should support this, as there are a few ways to achieve this already:

1. Putting standard config file in the project root that [extends](https://eslint.org/docs/user-guide/configuring#extending-configuration-files) the shared config in the Git submodule. This is how shared configs installed via `npm` also work.
1. Using the [--config](https://eslint.org/docs/user-guide/command-line-interface#-c-config) CLI flag.

**Comment by aladdin-add** [comment]
emm.. I've misread the issue somehow, please disregard my comment. 🤣

**Comment by christiantjl** [comment]
Hey everyone, thank you for your comments and suggested workarounds. This definitely isn't a needed feature, I would just find it super convenient to add my ESLint config to repos with version control using `git submodule add`, and this could be extended further with GitHub templates so ESLint setup involves *zero* steps with a global install via `npm i -g eslint`. Atm, I don't see a way of setting up config without publishing it a config to NPM or manually copying it over.

Admittedly, I dream of a day when Travis, NPM, and Git all respect a dot folder called `.config/` so everyone doesn't have 18 dotfiles piling up in root (`.eslintrc`, `.travis.yml`, `.npmignore`, `.gitignore`, Rollup configs, Webpack configs, etc.).

**Comment by mdjermanovic** [comment]
`extends` can accept relative paths. Config file in a project could be as simple as:

```json
{
  "extends": ["./.eslint/.eslintrc"]
}
```
